src/Microsoft.AspNetCore.Routing/Tree/TreeRouter.cs

```diff
@@ -166,36 +166,37 @@ public VirtualPathData GetVirtualPath(VirtualPathContext context)
 
                 var treeEnumerator = new TreeEnumerator(root, tokenizer);
 
+                // Create a snapshot before processing the route. We'll restore this snapshot before running each
+                // to restore the state. This is likely an "empty" snapshot, which doesn't allocate.
+                var snapshot = context.RouteData.PushState(null, null, null);
```

Comment:
Which test in this PR exercises this part of the code? was difficult to find out..